src/tools/crushtool.cc

```diff
@@ -840,6 +891,35 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (add_rule) {
+    if (crush.rule_exists(rule_name)) {
+      cerr << "rule " << rule_name << " already exists" << std::endl;
+      return 0;
+    }
+    int r = crush.add_simple_ruleset(rule_name, rule_root, rule_type, rule_mode,
+      pg_pool_t::TYPE_REPLICATED, &err);
+    if (r < 0) {
+      cerr << err.str() << std::endl;
+      return r;
```

Comment:
i don't think we can use a negative number here, per http://pubs.opengroup.org/onlinepubs/009695399/functions/exit.html